electric-sql/electric
# chore(client): Move Prisma and our generator to be dependencies of the electric client.

- Prisma is pinned to a known working version.
- Rather than `npx prisma` we resolve the prisma cli script that is a direct dependancy of our client package. This should enable projects using the electric client to install a different version of prisma to the one we use.

### To test:

Build the client:
```sh
cd client/typescript
pnpm install
pnpm build && pnpm pack
```
Update and test example:
```sh
cd ../../examples/[an example]
rm -rf node_modules/
rm package-lock.json  # If there is one
npm uninstall @prisma/client @electric-sql/prisma-generator prisma electric-sql
npm install ../../clients/typescript/electric-sql-0.8.0.tgz
```

Ensure ./backend/compose/.envrc is using the 0.8 tagged Electric Docker image:
```sh
# ./backend/compose/.envrc
export ELECTRIC_IMAGE=electricsql/electric:0.8
```

Then start the example as usual.

---

To do:

- [ ] Test examples - prisma and generator dependencies need removing:
  - [x] Linearlite **working**
  - [ ] expo
  - [x] web-wa-sqlite
  - [x] beer-starts
  - [x] inonic-demo
- [x] Test that a project can have a conflicting version of Prisma installed

After this is released the examples and starter will need updating to remove the Prisma and generator dependancies.
